tests/xscanalyticsevent_test.go

```diff
@@ -0,0 +1,62 @@
+package tests
+
+import (
+	"github.com/jfrog/jfrog-client-go/auth"
+	"github.com/jfrog/jfrog-client-go/http/jfroghttpclient"
+	"github.com/jfrog/jfrog-client-go/xsc/services"
+	"github.com/stretchr/testify/assert"
+	"regexp"
+	"testing"
+)
+
+var testsEventService *services.AnalyticsEventService
+
+func TestXscAddAndUpdateGeneralEvent(t *testing.T) {
+	xscDetails, client := initXscEventTest(t)
+	testsEventService = services.NewAnalyticsEventService(client)
+	testsEventService.XscDetails = xscDetails
+
+	event := services.XscAnalyticsBasicGeneralEvent{
+		EventType:              1,
+		EventStatus:            "started",
+		Product:                "cli",
+		ProductVersion:         "2.53.1",
+		IsDefaultConfig:        false,
+		JfrogUser:              "gail",
+		OsPlatform:             "mac",
+		OsArchitecture:         "arm64",
+		MachineId:              "id",
+		AnalyzerManagerVersion: "1.1.1",
+	}
+	msi, err := testsEventService.AddGeneralEvent(services.XscAnalyticsGeneralEvent{XscAnalyticsBasicGeneralEvent: event})
+	assert.NoError(t, err)
+	assert.True(t, isValidUUID(msi))
+
+	event = services.XscAnalyticsBasicGeneralEvent{
+		EventStatus:          "started",
+		TotalFindings:        10,
+		TotalIgnoredFindings: 5,
+		TotalScanDuration:    "15s",
+	}
+
+	err = testsEventService.UpdateGeneralEvent(services.XscAnalyticsGeneralEventFinalize{XscAnalyticsBasicGeneralEvent: event, MultiScanId: msi})
+	assert.NoError(t, err)
+}
+
+func isValidUUID(str string) bool {
+	uuidRegex := regexp.MustCompile(`^[0-9a-f]{8}-[0-9a-f]{4}-1[0-9a-f]{3}-[0-9a-f]{4}-[0-9a-f]{12}$`)
+	return uuidRegex.MatchString(str)
+}
+
+func initXscEventTest(t *testing.T) (xscDetails auth.ServiceDetails, client *jfroghttpclient.JfrogHttpClient) {
```

Comment:
There is an issue I noticed here that if the url provided as a flag doesn't have a suffix of '/xsc', nothing adds it to the url. This may cause a 404 status code to return from the POST query. Do we trust the customer to provide the correct url with the /xsc suffix (since the flag that is suppose to be provided is -url.xsc and not -jfrog.url)?

Reply:
like any other server (rt/dist/xray) if the user want to run tests for this server he has to provide the relevant url with the relevant suffix.